mono/metadata/appdomain.c

```diff
@@ -2597,9 +2597,14 @@ unload_thread_main (void *arg)
 
 	internal = mono_thread_internal_current ();
 
-	MonoString *thread_name_str = mono_string_new_checked (mono_domain_get (), "Domain unloader", &error);
-	if (is_ok (&error))
-		mono_thread_set_name_internal (internal, thread_name_str, TRUE, FALSE, &error);
+// "Domain unloader"
+#define THREAD_NAME 'D','o','m','a','i','n',' ', \
```

Comment:
Can't say I like how this looks, especially how repetitive it is. Can't we abstract this away in `mono_thread_set_name_internal`?

Reply:
This is the most efficient portable way.

Reply:
I mean, setting a thread name is not something one would do in a hot code path anyway. In fact, you're only allowed to set the name of a managed `Thread` object once. I'm really not concerned about performance for this case.

What's the portability concern? Can't we just pass a C string and convert it to UTF-16 as required?

Reply:
That was there, and then concerns to check errors, so this cuts that all out and requires utf16 input. Portability as in more readable L"foo" only is right sometimes.

Reply:
It looks like in your current version of `mono_thread_set_name_internal`, you have a code path that does UTF-8 -> UTF-16 conversion with error checking anyway, and you also have to convert to UTF-8 for the profiler callback, so why not just standardize on UTF-8 input? It would make all these call sites much cleaner at the cost of one string conversion which, as far as I can tell, the old code was doing through `mono_string_new_checked` anyway.

Reply:
I thought I trimmed that some. I'll look again. Almost anything is possible and I realize it isn't a hot path, but I'd prefer to reduce conversions, copying, and error paths in general, and demonstrate such, when it isn't too onerous. I agree the alternating ',' isn't ideal but I thought it acceptable. Later language standards do offer like u2"foo" or somesuch, we might use eventually.

Also I finally added proper kernel-based thread names in Windows 10 with a few advantages over the wierd old debugger-based mechanism, including that it is unicode-only, so utf8 has a little loss of canonicalness (and survives debugger absence, has an API to get the name, etc.)

Anyway I can go back a revision or two if you insist.